### PR TITLE
KOGITO-1830: BDD tests: Test service discovery on Spring Boot applica…

### PR DESCRIPTION
**Pull request by Sgitario**
…tion

JIRA Ticket: https://issues.redhat.com/browse/KOGITO-1830
Description: Provide spring boot version for onboarding example project. Actions taken:

- old onboarding to _onboarding-quarkus_ namings
- add new Spring Boot version as _onboarding-springboot_ namings

so we keep consistency with what we have at the root folder, both quarkus and springboot.

Tested with automation framework and following the instructions from README.

### CODE REVIEW

File: onboarding-example/onboarding-springboot/.gitignore [hunk 1]
```diff
@@ -0,0 +1,34 @@
+bin/
```
Comment (radtriste) [inline onboarding-example/onboarding-springboot/.gitignore]: That one is not mandatory I think and can even be deleted into onboarding-quarkus IMO

File: onboarding-example/pom.xml [hunk 2]
```diff
@@ -38,6 +40,25 @@
   <modules>
     <module>hr</module>
     <module>payroll</module>
-    <module>onboarding</module>
   </modules>
+
+  <profiles>
```
Comment (radtriste) [inline onboarding-example/pom.xml]: Isn't it a bit too much ?
I think you can add both as modules directly

**Review by radtriste — COMMENTED** [review]
I would update:
- old `onboarding` to `onboarding-quarkus` namings
- set the Spring Boot to `onboarding-springboot` namings

so we keep consistency with what we have at the root folder, both quarkus and springboot.
Then, for 0.10+ documentation, paths will need to be updated, cc @sterobin

File: onboarding-example/onboarding-springboot/.gitignore (hunk 1 above)
Comment (Sgitario) [reply onboarding-example/onboarding-springboot/.gitignore]: Removed from here and from onboarding-quarkus.

File: onboarding-example/pom.xml (hunk 2 above)
Comment (Sgitario) [reply onboarding-example/pom.xml]: Okis, removed the profiles and moved the child modules to the parent.

**Comment by Sgitario** [comment]
@radtriste about your comment above, I guess you meant to specify the changes in the JIRA description about what was done in this ticket. I already did this and also added the component "Documentation" into the JIRA, so it can be used to track the required changes within the documentation as well.

File: onboarding-example/onboarding-springboot/.gitignore (hunk 1 above)
Comment (radtriste) [reply onboarding-example/onboarding-springboot/.gitignore]: I was talking about the whole file :)

Comment (Sgitario) [reply onboarding-example/onboarding-springboot/.gitignore]: removed both files

**Review by sutaakar — APPROVED** [review]

**Review by radtriste — APPROVED** [review]

**Review by evacchi — APPROVED** [review]

**Comment by radtriste** [comment]
@cristianonicolai @evacchi can we merge that one if you are good with it ?

**Comment by evacchi** [comment]
@radtriste yes, waiting for green?

**Comment by radtriste** [comment]
yes sure :)

**Comment by radtriste** [comment]
@cristianonicolai @evacchi is green :)

File: onboarding-example/onboarding-springboot/pom.xml [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+      </activation>
+      <dependencies>
+        <dependency>
+          <groupId>org.kie.kogito</groupId>
+          <artifactId>infinispan-persistence-addon</artifactId>
+        </dependency>
+        <dependency>
+          <groupId>org.infinispan</groupId>
+          <artifactId>infinispan-spring-boot-starter-remote</artifactId>
+          <version>2.1.7.Final</version>
```
Comment (cristianonicolai) [inline onboarding-example/onboarding-springboot/pom.xml]: @Sgitario please remove the version from here, it should be managed at the root pom.

**Review by cristianonicolai — COMMENTED** [review]
@Sgitario could you also add copywrite headers to Java classes please.

File: onboarding-example/onboarding-springboot/pom.xml (hunk 3 above)
Comment (Sgitario) [reply onboarding-example/onboarding-springboot/pom.xml]: Removed. I've added the headers to all java sources that I created or moved as part of this PR.
I ran the new project using the persistence profile to ensure it's still working ok.

**Review by cristianonicolai — APPROVED** [review]

**Pull request merged**